### ISSUE TITLE
AlexNet performance from scratch

### ISSUE THREAD

**Issue by jwyang** [mahyarnajibi/fast-rcnn-torch#8]
Hi, Mahyar,

Thanks very much for sharing this code. It is very helpful!

I am wondering whether you tried to train the AlexNet (without BN) from scratch. What performance can you get? Recently, I can only get ~34% after 40K iterations.

Looking forward to your reply

thanks

**Comment by mahyarnajibi** [comment]
Hi,
Actually, I did not train the network totally from scratch. I used imagenet models for weight initialization. With VGG16 and by fine-tuning from imagenet model on VOC 2007 trainval, I got 67.2% mAP on VOC2007 test set after 40K iterations.

**Comment by jwyang** [comment]
Hi,

You mean you used pre trained AlexNet obtaining 67.2 on VOC2007? That's so
high! But I can get 56.6% which is coordinated with the statistics. Can you
explain more about this?

**Comment by mahyarnajibi** [comment]
Sorry, the result I reported was for VGG16 not AlexNet.

**Comment by jwyang** [comment]
Ok, So I am curious about training from scratch, since I can only get 33%
as high on VOC 2007. However, many papers reported their performance around
43%. What do you think is the reason ?

**Comment by mahyarnajibi** [comment]
Since the model can get almost the same mAP from pre-trained imagenet, probably there should not be a problem with the gradient computation during training. The reason might be the initialization of the layer weights when you start from scratch. Is your weight initialization for conv and fc layers the same as the 43% model?

**Comment by mahyarnajibi** [comment]
Also as suggested by the Fast R-CNN paper, I freezed the first convolutional layers in the lua code. Did you unfreeze them?

**Comment by jwyang** [comment]
Yes, I initialize the weights according to the papers as possible as I can.
And I tried many settings, however I failed. Do you think it is also
possible because the learning policy and data preprocessing? Meanwhile, I
will try to train a scratch model based on the caffe model to see what is
the performance.

**Comment by mahyarnajibi** [comment]
Did you also remove "conv1.accGradParameters = function() end" from the AlexNet model file? Currently, like the Fast R-CNN, this code supports random flipping as the only data augmentation technique and the default learning rate policy is the simple step function. Since all the settings are the same as the caffe implementation, comparison with the caffe-based implementation would be a good idea.

**Comment by jwyang** [comment]
yes, I removed it for fully training and get 33%. I guess I will pay some
time to compare with caffe code. If I found something, will let you know.

Thanks a lot for your reply!

**Comment by mahyarnajibi** [comment]
Sure, thanks!
